### PR TITLE
Fix mirror checkpointer error on the alter database query

### PR DESCRIPTION
**Pull request by andr-sokolov**
Steps to reproduce:
1. Set the fsync GUC to on.
2. Make changes to a database and move it to another tablespace, for example:
```
psql -c "create tablespace ts1 location '/tmp/ts'" postgres
psql -c "create database test" postgres
psql -c "create table t (i int)  DISTRIBUTED randomly" test
psql -c "alter database test SET TABLESPACE ts1" postgres
```

If a mirror checkpointer doesn't create checkpoint between "create table" and "alter database", then attempt to create checkpoint the next time will result in the file synchronization error, because the file doesn't exist. At the moment of
checkpoint creating the database file has already been moved to a new tablespace, but fsync request to checkpointer was not changed.

The error doesn't manifest itself in PostgreSQL. PostgreSQL removes files from the old tablespace and adds XLOG_DBASE_DROP record to WAL at the end of movedb(). GPDB has different implementation of movedb(). It schedules removing database directory on transaction commit. When PostgreSQL replica deletes old files, it processes the XLOG_DBASE_DROP record and calls the ForgetDatabaseSyncRequests() function. When GPDB mirror deletes old files, it
processes XLOG_XACT_COMMIT_PREPARED and doesn't change sync requests.

In the end of copydir() function the database is synced, so it is necessary to cancel all sync requests to checkpointer related to this database before the database files will be deleted.

### CODE REVIEW

File: src/backend/catalog/storage_database.c [hunk 1]
```diff
@@ -164,6 +165,14 @@ static void
 dropDatabaseDirectory(DbDirNode *deldb, bool isRedo)
 {
 	char *dbpath = GetDatabasePath(deldb->database, deldb->tablespace);
+
```
Comment (haolinw) [inline src/backend/catalog/storage_database.c:168]: Did you try to keep the same scenario as `dbase_redo`? It seems better to align with that if it also works.

```c
        /* Drop pages for this database that are in the shared buffer cache */
		DropDatabaseBuffers(xlrec->db_id);

		/* Also, clean out any fsync requests that might be pending in md.c */
		ForgetDatabaseSyncRequests(xlrec->db_id);

		/* Clean out the xlog relcache too */
		XLogDropDatabase(xlrec->db_id);

		/* And remove the physical files */
		if (!rmtree(dst_path, true))
			ereport(WARNING,
					(errmsg("some useless files may be left behind in old database directory \"%s\"",
							dst_path)));

```

Comment (andr-sokolov) [reply src/backend/catalog/storage_database.c:171]: @haolinw , Done

Comment (haolinw) [reply src/backend/catalog/storage_database.c:171]: Thanks @andr-sokolov . LGTM. By the way, seems the pipeline wasn't triggered automatically, could you please re-trigger it manually ? If everything works fine I think it's good to merge.

Comment (andr-sokolov) [reply src/backend/catalog/storage_database.c:171]: @haolinw , All checks have passed

**Review by haolinw — APPROVED** [review]

**Pull request merged**